### PR TITLE
Overhaul New Idea template plus BBcode support

### PR DESCRIPTION
**Pull request by VSEphpbb**

### CODE REVIEW

**Comment by rxu** [comment]
It looks like the stuff fixed here is still broken. F.e. statuses aren't displayed, a bunch of template vars is not defined, like all the ```*_STATUS```, ```S_SHOW_SMILEY_LINK``, as well as draft and review related vars.
Also, indentations are missing so the original code is hard to read and understand.

**Comment by VSEphpbb** [comment]
The undefined stuff are fine as we don't have drafts and things like that, so that's not really broken. I also do not intentionally want to show the statuses because I don't think it really serves any purpose here to show the BBcodes are on, Images are one, etc... stuff. The intention is just to have BBCodes available.

I left out indentation changes to make the diff easier to see what has changed.

**Comment by rxu** [comment]
Then just delete it?

**Comment by VSEphpbb** [comment]
I actually prefer to keep it as it makes diff easier between our template and prosilver's which we need to do from time to time to see if we are still current with prosilver or if anything has changed. But yes, the tabbing needs to be adjusted ;)

**Comment by VSEphpbb** [comment]
Updated!

**Comment by rxu** [comment]
Works as expected :+1:

**Comment by VSEphpbb** [comment]
Updated again

**Comment by michaelcullum** [comment]
I don't know how posting is supposed to normally work in phpBB so my reviewing isn't going to be particularly helpful here I think.

**Comment by VSEphpbb** [comment]
Well, all that was done here was consolidating an extraneous template into its parent. If you look at the first commit and the 3rd commit, those are the only real changes here.

File: controller/post_controller.php [hunk 1]
```diff
@@ -63,9 +63,29 @@ public function post()
...
+			'S_SMILIES_ALLOWED'	=> $smilies_status,
+			'S_LINKS_ALLOWED'	=> $url_status,
+			'S_BBCODE_IMG'		=> $img_status,
+			'S_BBCODE_FLASH'	=> $flash_status,
+			'S_BBCODE_QUOTE'	=> true,
+
+			'BBCODE_STATUS'		=> $this->language->lang(($bbcode_status ? 'BBCODE_IS_ON' : 'BBCODE_IS_OFF'), '<a href="' . $this->helper->route('phpbb_help_bbcode_controller') . '">', '</a>'),
+			'IMG_STATUS'		=> ($img_status) ? $this->language->lang('IMAGES_ARE_ON') : $this->language->lang('IMAGES_ARE_OFF'),
+			'FLASH_STATUS'		=> ($flash_status) ? $this->language->lang('FLASH_IS_ON') : $this->language->lang('FLASH_IS_OFF'),
+			'URL_STATUS'		=> ($bbcode_status && $url_status) ? $this->language->lang('URL_IS_ON') : $this->language->lang('URL_IS_OFF'),
```
Comment (Elsensee) [inline controller/post_controller.php]: Why do you check here for `$bbcode_status` and not in the other lines? Why is the [url]-bbcode so special to you? Don't discriminate, please :sob:

**Comment by Elsensee** [comment]
Otherwise looks good :+1:

File: controller/post_controller.php (hunk 1 above)
Comment (VSEphpbb) [inline controller/post_controller.php]: This is just copy paste out of the posting editor.

Comment (Elsensee) [inline controller/post_controller.php]: Oh. Then why do we discriminate?

Comment (VSEphpbb) [inline controller/post_controller.php]: I'm sure there is a reason somewhere deep in the annals of phpBB's core

**Pull request merged**